src/mixpanel-lite.js

```diff
@@ -177,6 +177,22 @@
         });
     }
 
+    /**
+     * Register a set of super properties unless they are already added
+     * @param {object} data - JSON key/value pair
+     * @returns {void}
+     */
+    function register_once(data) {
+        unpersistedData = Object.keys(data = {}).reduce(function (acc, key) {
```

Comment:
I'm struggling to understand what's happening here... could this be refactored to improve readability?

Reply:
The idea is to only register data that hasn't already been registered. The reduce is building a new object out of the data based on which keys haven't been added to properties already. Then I'm registering that new object using the regular register function. To me, this is the clearest way to do that, but if you have a preference for another style I can try changing it.

Reply:
Sorry, a "bit" late to this. Chat GTP suggested the following:

```js
/**
 * Register a set of super properties unless they are already added.
 * @param {object} data - JSON key/value pair of properties to register.
 * @returns {void}
 */
function register_once(data) {
    var unpersistedData = {};

    // Check each property in the input data
    Object.keys(data).forEach(function(key) {
        // If the property does not exist in _properties, add it to unpersistedData
        if (!_properties[key]) {
            unpersistedData[key] = data[key];
        }
    });

    // Register the new properties that were not already present
    register(unpersistedData);
}
```